File: C++/Easy/WrongAnswer/occurrences-after-bigram_2.cpp
class Solution {
public:
    vector<string> findOcurrences(string text, string first, string second) {
        
        vector<string> res;
        istringstream iss(text);
        string s1, s2;
        while (iss >> s1) {
            if (s1 == first && s2 == second) {
                string s3;
                iss >> s3;
                res.push_back(s3);
            }
            s2 = s1;
        }
        return res;
    }
};

File: Rust/Easy/Accepted/occurrences-after-bigram_2.rs
impl Solution {
    pub fn find_ocurrences(text: String, first: String, second: String) -> Vec<String> {
        
        let mut res = Vec::new();
        let words = text.split(" ").collect::<Vec<&str>>();
        let n = words.len();
        for i in 0..n-2 {
            if words[i] == first && words[i+1] == second {
                res.push(words[i+2].to_string());
            }
        }
        res
    }
}